libraries/SD_MMC/examples/SDMMC_Test/SDMMC_Test.ino

```diff
@@ -1,21 +1,59 @@
+ * pin 4 - VDD (3.3V)        |                    |
+ * pin 5 - CLK               | 8 7 6 5 4 3 2 1   /
+ * pin 6 - VSS (GND)         | ▄ ▄ ▄ ▄ ▄ ▄ ▄ ▄  /
+ * pin 7 - D0                | ▀ ▀ █ ▀ █ ▀ ▀ ▀ |
+ * pin 8 - D1                |_________________|
+ *                             ║ ║ ║ ║ ║ ║ ║ ║
+ *                     ╔═══════╝ ║ ║ ║ ║ ║ ║ ╚═════════╗
+ *                     ║         ║ ║ ║ ║ ║ ╚══════╗    ║
+ *                     ║   ╔═════╝ ║ ║ ║ ╚═════╗  ║    ║
+ * Connections for     ║   ║   ╔═══╩═║═║═══╗   ║  ║    ║
+ * full-sized          ║   ║   ║   ╔═╝ ║   ║   ║  ║    ║
+ * SD card             ║   ║   ║   ║   ║   ║   ║  ║    ║
+ * ESP32-S3 DevKit  | 21  47  GND  39 3V3 GND  40 41  42  |
+ * ESP32-S3-USB-OTG | 38  37  GND  36 3V3 GND  35 34  33  |
+ * ESP32            |  4   2  GND  14 3V3 GND  15 13  12  |
+ * Pin name         | D1  D0  VSS CLK VDD VSS CMD D3  D2  |
+ * SD pin number    |  8   7   6   5   4   3   2   1   9 /
+ *                  |                                  █/
+ *                  |__▍___▊___█___█___█___█___█___█___/
+ * WARNING: ALL data pins must be pulled up to 3.3V with external 10k Ohm resistor!
+ * Note to ESP32 pin 2 (D0) : add 1K pull up after flashing
  *
- * SD Card | ESP32
- *    D2       12
- *    D3       13
- *    CMD      15
- *    VSS      GND
- *    VDD      3.3V
- *    CLK      14
- *    VSS      GND
- *    D0       2  (add 1K pull up after flashing)
- *    D1       4
+ * For more info see file README.md in this library or on URL:
+ * https://github.com/espressif/arduino-esp32/tree/master/libraries/SD_MMC
  */
 
 #include "FS.h"
 #include "SD_MMC.h"
 
+// Default pins for ESP-S3
+// Warning: ESP32-S3-WROOM-2 is using most of the default SD_MMC GPIOs (33-37) to interface with on-board OPI flash.
+//   If the SD_MMC is initialized with default pins it will result in rebooting loop - please
+//   reassign the pins elsewhere using the mentioned command `setPins`.
+// Note: ESP32-S3-WROOM-1 does not have GPIO 33 and 34 broken out.
+// Note: The board SP32-S3-USB-OTG has predefined default pins and the following definitions with the setPins() call will not be compiled.
+// Note: Pins in this definition block are ordered from top down in order in which they are on the full-sized SD card
+//       from left to right when facing the pins down (when connected to a breadboard)
+
+#if defined(SOC_SDMMC_USE_GPIO_MATRIX) && not defined(BOARD_HAS_SDMMC)
+    int d1  = 21; // SD pin 8 - add 10k Pullup to 3.3V if using 4-bit mode (use_1_bit_mode = false)
+    int d0  = 47; // SD pin 7 - add 10k Pullup
+    //     GND pin - SD pin 6
+    int clk = 39; // SD pin 5 - add 10k Pullup
+    //    3.3V pin - SD pin 4
+    //     GND pin - SD pin 3
+    int cmd = 40; // SD pin 2 - add 10k Pullup
+    int d3  = 41; // SD pin 1 - add 10k Pullup to card's pin even when using 1-bit mode
+    int d2  = 42; // SD pin 9 - add 10k Pullup to 3.3V if using 4-bit mode (use_1_bit_mode = false)
```

Comment:
```suggestion
    int d1  = 21; // SD pin 8 -Add 10k Ohm pull-up resistor if using 4-bit mode (use_1_bit_mode = false)
    int d0  = 47; // SD pin 7 - Add 10k Ohm pull-up resistor
    //     GND pin - SD pin 6
    int clk = 39; // SD pin 5 - Add 10k Ohm pull-up resistor
    //    3.3V pin - SD pin 4
    //     GND pin - SD pin 3
    int cmd = 40; // SD pin 2 - Add 10k Ohm pull-up resistor
    int d3  = 41; // SD pin 1 - Add 10k Ohm pull-up resistor to card's pin even when using 1-bit mode
    int d2  = 42; // SD pin 9 - Add 10k Ohm pull-up resistor if using 4-bit mode (use_1_bit_mode = false)
```

Reply:
I want to keep the 3.3V mention, otherwise, the users will connect it to 5V. Otherwise, I accept the suggestions but will commit from the command line.

Reply:
Ah, yeah. I forgot about the 5V pin. IMHO I think it's best to add a note before or add the mention to all lines. Something like "Add 10k Ohm pull-up resistor to 3.3V".

Reply:
And sorry about all the nit-picks 😥 
I'm used to this kind of review on NuttX. I'll try to be more objective on reviews,.

Reply:
I have added the 3.3 mention to all pullup comment in this file in separate commit.